modules/identity/src/apiGateway.ts

```diff
@@ -0,0 +1,85 @@
+import { ValidationError } from '@modules/errors';
+import type { Stage } from '@modules/stage';
+import type { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import {
+	ExpiredTokenError,
+	InvalidScopesError,
+	InvalidTokenError,
+	OktaTokenHelper,
+} from '@modules/identity/identity';
+
+export type AuthenticatedApiGatewayEvent = APIGatewayProxyEvent & {
+	type: 'AuthenticatedApiGatewayEvent';
+	identityId: string;
+};
+
+export type FailedAuthenticationResponse = APIGatewayProxyResult & {
+	type: 'FailedAuthenticationResponse';
+};
+
+export class IdentityApiGatewayAuthenticator {
+	tokenHelper: OktaTokenHelper;
+	constructor(stage: Stage, requiredScopes: string[]) {
+		this.tokenHelper = new OktaTokenHelper(stage, requiredScopes);
+	}
+
+	async authenticate(
+		event: APIGatewayProxyEvent,
+	): Promise<AuthenticatedApiGatewayEvent | FailedAuthenticationResponse> {
+		const authHeader = event.headers.Authorization;
+		if (!authHeader) {
+			console.log('No Authorization header provided in request', event);
+			return {
+				type: 'FailedAuthenticationResponse',
+				statusCode: 403,
+				body: JSON.stringify({ message: 'No Authorization header provided' }),
+			};
```

Comment:
No authorization header should be a 401

```suggestion
			return {
				type: 'FailedAuthenticationResponse',
				statusCode: 401,
				body: JSON.stringify({ message: 'No Authorization header provided' }),
			};
```